crates/storage/provider/src/transaction.rs

```diff
@@ -1634,8 +1527,32 @@ mod test {
         assert_eq!(take, vec![(block1.clone(), exec_res1.clone())]);
         assert_genesis_block(&tx, genesis.clone());
 
-        tx.insert_block(block1.clone(), &chain_spec, exec_res1.clone()).unwrap();
-        tx.insert_block(block2.clone(), &chain_spec, exec_res2.clone()).unwrap();
+        exec_res1.clone().write_to_db(tx.deref_mut(), 0).unwrap();
+        tx.insert_block(block1.clone()).unwrap();
+        tx.insert_hashes(
```

Comment:
Main reaction here is that we used to call the function insert_block and now it only inserts the historical block indices, so we'll want to rename it into something more descriptive I think. And we may want to have a function around which does all 3 operations?

Reply:
We would never use the one that does all 3 in any part of main reth, maybe only in tests - but then again, what would you be testing in that case?